controllers/argocd/util.go

```diff
@@ -1025,6 +1025,30 @@ func (r *ReconcileArgoCD) setResourceWatches(bldr *builder.Builder, clusterResou
 		},
 	}
 
+	// Add new predicate to delete Notifications Resources. The predicate watches the Argo CD CR for changes to the `.spec.Notifications.Enabled`
+	// field. When a change is detected that results in notifications being disabled, we trigger deletion of notifications resources
+	// repoDeployment := predicate.Funcs{
+	// 	UpdateFunc: func(e event.UpdateEvent) bool {
+	// 		newCR, ok := e.ObjectNew.(*argoprojv1a1.ArgoCD)
+	// 		if !ok {
+	// 			return false
+	// 		}
+	// 		oldCR, ok := e.ObjectOld.(*argoprojv1a1.ArgoCD)
+	// 		if !ok {
+	// 			return false
+	// 		}
+	// 		if (oldCR.Spec.Repo.ServiceAccount != newCR.Spec.Repo.ServiceAccount) || (!oldCR.Spec.Repo.MountSAToken && newCR.Spec.Repo.MountSAToken) {
+	// 			useTLSForRedis := r.redisShouldUseTLS(newCR)
+	// 			err := r.reconcileRepoDeployment(newCR, useTLSForRedis)
+	// 			if err != nil {
+	// 				log.Error(err, fmt.Sprintf("Failed to delete notifications controller resources for ArgoCD %s in namespace %s",
+	// 					newCR.Name, newCR.Namespace))
+	// 			}
+	// 		}
+	// 		return true
+	// 	},
+	// }
+
```

Comment:
@rishabh625 This looks like an unintended change

Reply:
Yeah Thanks @jaideepr97  good catch, i almost forgot that I commented it.